src/zenml/steps/base_step.py

```diff
@@ -169,19 +167,6 @@ def __init__(
 
         name = name or self.__class__.__name__
 
-        requires_context = self.entrypoint_definition.context is not None
```

Comment:
Shouldn't this be kept for backwards compatibility? Right now the behaviour would change for steps that still have the `StepContext` annotation I think.